src/server/worker.cc

```diff
@@ -76,17 +77,22 @@ Worker::Worker(Server *srv, Config *config) : srv(srv), base_(event_base_new())
 }
 
 Worker::~Worker() {
-  std::vector<redis::Connection *> conns;
-  conns.reserve(conns_.size() + monitor_conns_.size());
-
-  for (const auto &iter : conns_) {
-    conns.emplace_back(iter.second);
-  }
-  for (const auto &iter : monitor_conns_) {
-    conns.emplace_back(iter.second);
+  {
+    std::lock_guard<std::mutex> guard(conns_mu_);
+    for (const auto &iter : conns_) {
+      if (ConnMap::accessor accessor; conns_.find(accessor, iter.first)) {
+        accessor->second->Close();
```

Comment:
I don't know whether close the connection in the critical section is really a good idea...

Reply:
Hi @mapleFU ,

Sorry for putting time-costly tasks into a critical section.
I will make this action only manipulate with the container and leave `Close()` action outside the critical section.

Best Regards,
Edward

Reply:
It's ok, we can use the method here: https://github.com/apache/kvrocks/pull/2461#discussion_r1701976227 or using a vector to collect the conns and close them after lock. Generally, IO related options should be better outside critical section lock for performance

Reply:
Hi @mapleFU ,
Sorry for delay reply.
Thanks for your patience and suggestions!

I have updated code with `tbb::parallel_reduce` to collect the connections and close them out of tbb loop.

Best Regards,
Edward